src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -0,0 +1,109 @@
+            """)
+    Integer getActiveUsers(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select count(e.id)
+            from Exercise e
+            where e.releaseDate >= :#{#span} and e.releaseDate <= :#{#now}
+            """)
+    Integer getReleasedExercises(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(e.id)
+            from Exercise e
+            where e.dueDate >= :#{#span} and e.dueDate <= :#{#now}
+            """)
+    Integer getExerciseDeadlines(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(e.id)
+            from Exam e
+            where e.endDate >= :#{#span} and e.endDate <= :#{#now}
+            """)
+    Integer getConductedExams(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(se.id)
+            from StudentExam se, Exam e
+            where se.submitted = true and se.exam = e and e.endDate >= :#{#span}
+            """)
+    Integer getExamParticipations(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select sum(e.registeredUsers.size)
+            from Exam e
+            where e.endDate >= :#{#span} and e.endDate <= :#{#now}
+            """)
+    Integer getExamRegistrations(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(distinct r.assessor.id)
+            from Result r
+            where (r.assessmentType = 'MANUAL' or r.assessmentType = 'SEMI-AUTOMATIC') and r.completionDate >= :#{#span}
+            """)
+    Integer getActiveTutors(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select count(r.id)
+            from Result r
+            where r.completionDate >= :#{#span}
+            """)
+    Integer getCreatedResults(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select sum(r.feedbacks.size)
+            from Result r
+            where r.completionDate >= :#{#span}
+            """)
+    Integer getResultFeedbacks(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select s.submissionDate as day, count(s.id) as amount
+            from Submission s
+            where s.submissionDate > :#{#startDate}
+            group by s.submissionDate
+            order by s.submissionDate asc
+            """)
+    List<Map<String, Object>> getTotalSubmissions(ZonedDateTime startDate);
+
+    @Query("""
+            select s.submissionDate as day, count(s.id) as amount
+            from Submission s
+            where s.submissionDate > :#{#startDate} and s.submissionDate < :#{#endDate}
+            group by s.submissionDate
+            order by s.submissionDate asc
+            """)
+    List<Map<String, Object>> getTotalSubmissionsYesterday(ZonedDateTime startDate, ZonedDateTime endDate);
```

Comment:
Why does the method name contain "yesterday" when there are date range paramters?

Reply:
Also deprecated, this was used to extract the submission from yesterday, not needed atm. Will be refactored